### DISCUSSION TITLE
Support all CSL JSON fields in `citation:` CSL YAML.

### DISCUSSION THREAD

**Discussion by the-solipsist** [quarto-dev/quarto-cli#6766] [Feature Requests]
### Summary
Currently, some valid CSL JSON fields (such as [`id:`](https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#id-field)) are not supported in the CSL YAML within `citation:`.

### What I'm trying to do
I have all of the articles I've written saved in Zotero. I've exported all of them into a single CSL YAML file using BetterBibTex.  I then cut-paste the relevant CSL YAML snippet into the `citation:` field of the corresponding article's .qmd file.

### Problem
When I do so and click `render`, the YAML checker spits out an error that `id` is not a valid field:

```
[{
	"resource": "/home/sol/documents/work/website/publications/press/whatsapp-in-kashmir_when-big-brother-wants-to-go-beyond-watching-you.qmd",
	"owner": "_generated_diagnostic_collection_name_#1",
	"severity": 8,
	"message": "property name id is invalid",
	"source": "quarto",
	"startLineNumber": 14,
	"startColumn": 3,
	"endLineNumber": 14,
	"endColumn": 5
}]
```

Further, it doesn't recognize it when [authors and dates are given as arrays](https://github.com/quarto-dev/quarto-cli/issues/6680) instead of strings, but in that case it just ignores them.

### Proposed solution
Ensure that valid CSL YAML citations generated by Pandoc / Zotero BetterBibTex can be read by Quarto by supporting all valid CSL JSON fields.

**Comment by cderv** [comment]
@dragonstyle do we omit some fields from spec from some reason ?

**Reply by dragonstyle** [reply]
No this was merely an oversight. I just fixed it here:

https://github.com/quarto-dev/quarto-cli/commit/2e7ba02abf92f3ef01d3fb9fdbc8e1734c10f8be

We intend to support the full CSL JSON schema, so when there is an omission it is helpful to list it (as this thread did with `id` - thank you!).
